src/Temporalio/Bridge/src/runtime.rs

```diff
@@ -32,7 +39,36 @@ pub struct TelemetryOptions {
 #[repr(C)]
 pub struct LoggingOptions {
     filter: ByteArrayRef,
-    forward: bool,
+    /// This callback is expected to work for the life of the runtime.
+    forward_to: ForwardedLogCallback,
+}
+
+// This has to be Option here because of https://github.com/mozilla/cbindgen/issues/326
+/// Operations on the log can only occur within the callback, it is freed
+/// immediately thereafter.
+type ForwardedLogCallback =
+    Option<unsafe extern "C" fn(level: ForwardedLogLevel, log: *const ForwardedLog)>;
+
+pub struct ForwardedLog {
+    core: CoreLog,
+    fields_json: Arc<Mutex<Option<String>>>,
+}
+
+#[repr(C)]
+pub struct ForwardedLogDetails {
+    target: ByteArray,
+    message: ByteArray,
+    timestamp_millis: u64,
+    fields_json: ByteArray,
```

Comment:
I'm confused why we're converting the fields to JSON here rather than passing something structured back into .NET? Can we not convert into some .NET object here and send that over in `ForwardedLog` rather than doing this serialization?

Reply:
To pass something structured over the C FFI boundary means I build an entire object model in C to represent that. This usually means a recursive union with a new enum. It's just more complicated, I don't already have `serde_json::Value` at my disposal in C. It's a similar question to why I send proto bytes to/from core instead of creating structures to represent those.

But if it's worth it, I can do it. For now, I'm just converting to `Dictionary<string, JsonElement>` on the .NET side anyways.

Reply:
Yeah, I get that - I just thought it would probably be easy enough to make the right structure here since there's only a handful of types. I think it's probably worth it in the sense that it's not code we're likely to have any problem with, just a bit of extra boilerplate but it'll be quite a bit more efficient on a fairly hot path.

Reply:
I don't think it's going to be that much more efficient since I am turning it back into a string anyways. So I guess I can just let the Rust side turn it into a string for me and I can return a collection of pairs w/ string key and JSON string value. I have tried to avoid building a map/list structure in the C header so far and have been successful until now.

Reply:
Ok, I updated this to give keys and values where the values are raw JSON. This is the best performing I can do, and is better than converting each nested Rust object into a nested .NET object just to turn it to a string in the log message on the .NET side. Having Rust just serialize JSON is plenty fast for this use case (and not worth a new C map structure).

Reply:
:+1: makes sense to me. What I missed originally is that `ILogger` doesn't have any explicit structured k/v pairs in the interface anyway, so yeah there's no "place" to offer the structured objects.

Reply:
> there's no "place" to offer the structured objects.

Well, there is in that a logger gets an arbitrary "state" which is an enumerable of k/v pairs. So advanced users can use that to get the fields out (we do in our tests). But the default is to append the field values stringified anyways. So I'd prefer to make that the performant path and force users to deserialize JSON if they need more advanced structure if they happen to be doing something advanced with the values.